src/main/java/org/broadinstitute/consent/http/service/dao/DraftServiceDAO.java

```diff
@@ -0,0 +1,156 @@
+package org.broadinstitute.consent.http.service.dao;
+
+import com.google.inject.Inject;
+import jakarta.ws.rs.BadRequestException;
+import jakarta.ws.rs.NotAuthorizedException;
+import jakarta.ws.rs.NotFoundException;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import org.broadinstitute.consent.http.db.DraftDAO;
+import org.broadinstitute.consent.http.enumeration.UserRoles;
+import org.broadinstitute.consent.http.models.DraftInterface;
+import org.broadinstitute.consent.http.models.DraftSummary;
+import org.broadinstitute.consent.http.models.FileStorageObject;
+import org.broadinstitute.consent.http.models.User;
+import org.glassfish.jersey.media.multipart.FormDataBodyPart;
+import org.jdbi.v3.core.Jdbi;
+
+public class DraftServiceDAO {
+
+  private final Jdbi jdbi;
+  private final DraftDAO draftDAO;
+  private final DraftFileStorageServiceDAO draftFileStorageServiceDAO;
+
+  @Inject
+  public DraftServiceDAO(Jdbi jdbi, DraftDAO draftDAO,
+      DraftFileStorageServiceDAO draftFileStorageServiceDAO) {
+    this.jdbi = jdbi;
+    this.draftDAO = draftDAO;
+    this.draftFileStorageServiceDAO = draftFileStorageServiceDAO;
+  }
+
+  public void insertDraft(DraftInterface draft)
+      throws SQLException, BadRequestException {
+    jdbi.useHandle(handle -> {
+      handle.getConnection().setAutoCommit(false);
+      try {
+        draftDAO.insert(draft.getName(), draft.getCreateDate().toInstant(),
+            draft.getCreateUser().getUserId(), draft.getJson(), draft.getUUID(),
+            draft.getType().getValue());
+      } catch (Exception e) {
+        handle.rollback();
+        throw new BadRequestException(
+            "Error submitting draft.  Drafts require valid json to be submitted.");
+      }
+      handle.commit();
+    });
+  }
+
+  public DraftInterface updateDraft(DraftInterface draft, User user) throws SQLException {
+    draft.setUpdateUser(user);
+    draft.setUpdateDate(new Date());
+    jdbi.useHandle(handle -> {
+      handle.getConnection().setAutoCommit(false);
+      try {
+        draftDAO.updateDraftByDraftUUID(draft.getName(),
+            draft.getUpdateDate().toInstant(), draft.getUpdateUser().getUserId(), draft.getJson(),
+            draft.getUUID(), draft.getType().getValue());
+      } catch (Exception e) {
+        handle.rollback();
+      }
+      handle.commit();
+    });
+    return getAuthorizedDraft(draft.getUUID(), user);
+  }
+
+  public DraftInterface getAuthorizedDraft(UUID draftUUID, User user) {
+    DraftInterface draft;
+    try {
+      draft = findDraftByDraftUUID(draftUUID);
+    } catch (SQLException e) {
+      throw new NotFoundException(
+          String.format("Draft with UUID %s not found.", draftUUID.toString()));
```

Comment:
arguments to `format()` already undergo `toString()` so it's unnecessary here
```suggestion
          String.format("Draft with UUID %s not found.", draftUUID));
```

Reply:
Surprised my IDE didn't hint this.  It has in the past.  I need to look at settings.  Good find.  Thanks!